Cannot release to GitHub using workflow token

### Description

GitHub Action workflow tokens cannot be used to authenticate to the GitHub REST API to create a new release.

### How To Reproduce

1. Bump the project version
2. Merge changes
3. Observe the error in the `release-to-github` step of the triggered `Publish` workflow

### Expected Behavior

The workflow should release the new package version

### Additional Context

API docs: https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28#create-a-release
Error log: https://github.com/SituDevelopment/entegy-sdk-python/actions/runs/8415423732/job/23040578601#step:3:7

### Log Output

```shell
2024-03-25T05:15:33.3857138Z ##[group]Run gh release create --generate-notes --target main "v7.0.0"
2024-03-25T05:15:33.3858389Z [36;1mgh release create --generate-notes --target main "v7.0.0"[0m
2024-03-25T05:15:33.3902032Z shell: /usr/bin/bash -e {0}
2024-03-25T05:15:33.3902515Z env:
2024-03-25T05:15:33.3903225Z   GH_TOKEN: ***
2024-03-25T05:15:33.3903685Z ##[endgroup]
2024-03-25T05:15:33.6148550Z HTTP 403: Resource not accessible by integration (https://api.github.com/repos/SituDevelopment/entegy-sdk-python/releases)
2024-03-25T05:15:33.6178190Z ##[error]Process completed with exit code 1.
```

--- a/.github/workflows/publish.yaml
+++ b/.github/workflows/publish.yaml
@@ -22,7 +22,7 @@ jobs:
   release-to-github:
     runs-on: ubuntu-latest
     env:
-      GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      GH_TOKEN: ${{ secrets.RELEASE_PAT }}
     if: ${{ fromJson(needs.check-version.outputs.is-greater) }}
     needs: check-version
     permissions:

